Financial-Times/n-myft-ui
# ug-931 new list design:

Jira ticket: [ug-931](https://financialtimes.atlassian.net/browse/UG-931)

This PR implements the following parts of the new list design (see [wireframe](https://www.figma.com/file/09mlE9M0XVG5fIZy99tls2/(Final)-Professor-Engagement?node-id=758%3A7466))

- The dialogue displays all the users existing lists, with the most recent one added at the top
- If the article is already contained within a list, the checkbox next to the list name is checked, otherwise it is unchecked

Screenshot:

![Screenshot 2022-07-22 at 15 36 53](https://user-images.githubusercontent.com/54366961/180462632-d3bc711b-e0cd-4123-b38f-2a5fd387e4e9.png)

To test this code:

- Check out this branch and run npm link
- Clone next-article and run npm link @financial-times/n-myft-ui
- Update next-myft-client on next-article to v10.3.0
- Run next-article with npm start
- Switch on the manageArticleLists flag on [Toggler](https://toggler.ft.com/#manageArticleLists)
- Got to a local article, e.g. https://local.ft.com:5050/content/cf5d9a52-f193-4c7a-b5d0-4182b6ddcf3d
- Click on "Save" and the manageArticleLists modal should appear.
- Select 'add to a new list' and create a list. It should appear in the list of lists, checked.
- Repeat and check that the most recent one created is at the top
- Close the modal and reload the page. Click on the 'saved' button. The lists that you previously added the article should still be checked.
